File: cleanup/cleanup.py
class Mess:
    def __init__(self, size):
        print(f"Make a {size} mess.")
        self.size = size

    def __enter__(self):
        return self

    def __exit__(self, *_):
        print(f"Clean up the {self.size} mess.")

    def look(self):
        print(f"Look at this {self.size} mess!")


with Mess("huge") as huge_mess:
    huge_mess.look()

File: cleanup/cleanup.rs
struct Mess {
    size: String,
}

impl Mess {
    fn new(size: String) -> Mess {
        println!("Make a {} mess.", size);
        Mess { size }
    }

    fn look(&self) {
        println!("Look at the {} mess!", self.size);
    }
}

impl Drop for Mess {
    fn drop(&mut self) {
        println!("Clean up the {} mess.", self.size);
    }
}

fn main() {
    let giant_mess = Mess::new("giant".into());
    giant_mess.look();
}
